### PR TITLE
tsv test pipeline

### PR DESCRIPTION
**Pull request by ckairen**
Adding pipeline to do civ test on the TypeSpecValidation tool

### CODE REVIEW

File: eng/tools/TypeSpecValidation/test/git-diff.test.ts [hunk 1]
```diff
@@ -7,5 +7,6 @@ describe("git-diff", function () {
     const result = await new GitDiffRule().execute(new TsvTestHost(), TsvTestHost.folder);
 
     assert(result.success);
+    assert(false);
```
Comment (ckairen) [inline eng/tools/TypeSpecValidation/test/git-diff.test.ts:10]:
```suggestion
```

**Comment by ckairen** [comment]
Tested failure for test here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3230024&view=results

File: eng/pipelines/typespec-validation-ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  pool:
+    name: $(Pool)
+    vmImage: $(OSVmImage)
+
+  steps:
+  - template: /eng/pipelines/templates/steps/npm-install.yml
+
+  - pwsh: |
+      npm test
```
Comment (mikeharder) [inline eng/pipelines/typespec-validation-ci.yml:39]:
```suggestion
  - script: npm test
```

File: eng/pipelines/typespec-validation-ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    displayName: Test TypeSpecValidation Tool
+    condition: succeededOrFailed()
```
Comment (mikeharder) [inline eng/pipelines/typespec-validation-ci.yml:41]:
```suggestion
```

I don't think we should run tests if install failed.

File: eng/pipelines/typespec-validation-ci.yml (hunk 2 above)
Comment (mikeharder) [reply eng/pipelines/typespec-validation-ci.yml:39]: Don't need powershell

**Review by mikeharder — APPROVED** [review]

**Pull request merged**